api/v1alpha1/apidefinition_webhook.go

```diff
@@ -202,14 +202,24 @@ func (in *ApiDefinition) ValidateDelete() error {
 
 func (in *ApiDefinition) validateTarget() field.ErrorList {
```

Comment:
Due to the relatively high cyclomatic complexity of this function, i think it might be a good idea to write a test covering each scenario.

Reply:
I've added test cases for `validateTarget`